pkg/reconciler/v1alpha1/pipelinerun/pipelinerun.go

```diff
@@ -464,6 +483,16 @@ func (c *Reconciler) createTaskRun(logger *zap.SugaredLogger, rprt *resources.Re
 	return c.PipelineClientSet.TektonV1alpha1().TaskRuns(pr.Namespace).Create(tr)
 }
 
+func addRetryHistory(task *v1alpha1.TaskRun) {
```

Comment:
This function does a little bit more than add history, it also resets the Status of the TaskRun.  Perhaps break this down into two functions - `addRetryHistory` and `clearStatus`?

Also suggest changing the argument name here from `task` to `taskRun` or `tr` for consistency.